src/Producer.cpp

```diff
@@ -0,0 +1,45 @@
+#include "Producer.h" 
+
+// Init static  
+Lockers Producer::_lockers;
+
+
+Producer::Producer(std::list<std::tuple<thread::id, double, bool>>& global_list, thread::id thread_id, int millisecondsToWait) :
+	_global_list(global_list), _thread_id(thread_id), _millisecondsToWait(millisecondsToWait){}
+
+
+
+void Producer::operator() () {
+
+	double value = std::hash<thread::id>{}(_thread_id);
+	int i = 2;
+	while (value > 0)
+	{
+		std::this_thread::sleep_for(std::chrono::milliseconds(500));  
+
+		{
+			std::unique_lock lock(_lockers._mtx_global_list);
+			_global_list.push_back(std::make_tuple(_thread_id, value, is_last_message));
+		}
+		_lockers._cond_global_list.notify_all();
+
+
+
+		value /= 10.0;  // Divide value by 10 for the next message
+		--i;
+	}
+
+
+
+	std::unique_lock lock(_lockers._mtx_global_list);
+	is_last_message = true;  // Set the last message flag
```

Comment:
you're not pushing this to the vector, so the threads never stop, you have an endless loop